Quantum-Control-Applications/Superconducting/Single Fixed Transmon/Use Case - icognito - Optimized readout with optimal weights/IQ_blobs_opt_weights_benchmark.py

```diff
@@ -0,0 +1,129 @@
+                                      config=config,
+                                      update_tof=False,
+                                      rr_qe=rr_qe,
+                                      path=f'ge_disc_params_{rr_qe}.npz',
+                                      lsb=lsb)
+
+with program() as benchmark:
+    n = declare(int)
+    I = declare(fixed)
+    Q = declare(fixed)
+    res = declare(bool)
+    I_st = declare_stream()
+    Q_st = declare_stream()
+    res_st = declare_stream()
+    adc_st = declare_stream(adc_trace=True)
+
+    with for_(n, 0, n < n_runs, n + 1):
+        discriminator.measure_state("readout", "out1", "out2", res=res, I=I, Q=Q)
+        # measure(
+        #     "readout",
+        #     "resonator",
+        #     adc_st,
+        #     dual_demod.full("cos", "out1", "sin", "out2", I),
+        #     dual_demod.full("minus_sin", "out1", "cos", "out2", Q),
+        # )
+        save(I, I_st)
+        save(Q, Q_st)
+        save(res, res_st)
+        wait(cooldown_time, "resonator")
+
+        align()  # global align
+
+        play("x180", "qubit")
+        align("qubit", "resonator")
+        discriminator.measure_state("readout", "out1", "out2", res=res, I=I, Q=Q)
+        # measure(
+        #     "readout",
+        #     "resonator",
+        #     adc_st,
+        #     dual_demod.full("cos", "out1", "sin", "out2", I),
+        #     dual_demod.full("minus_sin", "out1", "cos", "out2", Q),
+        # )
+        save(I, I_st)
+        save(Q, Q_st)
+        save(res, res_st)
+        wait(cooldown_time, "resonator")
+
+        seq0 = [0, 1] * n_runs
+
+    with stream_processing():
+        res_st.boolean_to_int().save_all('res')
+        I_st.save_all('I')
+        Q_st.save_all('Q')
+
+qm = qmm.open_qm(config)
+
+job = qm.execute(benchmark)
+
+result_handles = job.result_handles
+result_handles.wait_for_all_values()
+res = result_handles.get('res').fetch_all()['value']
+I = result_handles.get('I').fetch_all()['value']
+Q = result_handles.get('Q').fetch_all()['value']
+
+plt.figure()
+plt.hist(I[np.array(seq0) == 0], 50)
+plt.hist(I[np.array(seq0) == 1], 50)
+plt.plot([discriminator.get_threshold()] * 2, [0, 60], 'g')
+plt.show()
+
+plt.figure()
+plt.plot(I, Q, '.')
+theta = np.linspace(0, 2 * np.pi, 100)
+for i in range(discriminator.num_of_states):
+    a = discriminator.sigma[i] * np.cos(theta) + discriminator.mu[i][0]
+    b = discriminator.sigma[i] * np.sin(theta) + discriminator.mu[i][1]
+    plt.plot([discriminator.mu[i][0]], [discriminator.mu[i][1]], 'o')
+    plt.plot(a, b)
+plt.axis('equal')
+
+p_s = np.zeros(shape=(2, 2))
+for i in range(2):
+    res_i = res[np.array(seq0) == i]
+    p_s[i, :] = np.array([np.mean(res_i == 0), np.mean(res_i == 1)])
+
+labels = ['g', 'e']
+plt.figure()
+ax = plt.subplot()
+sns.heatmap(p_s, annot=True, ax=ax, fmt='g', cmap='Blues')
+
+ax.set_xlabel('Predicted labels')
+ax.set_ylabel('Prepared labels')
+ax.set_title('Confusion Matrix')
```

Comment:
I would name it fidelity. See the names I gave in the other ge_discriminator

Reply:
did!